### ISSUE TITLE
Integration Testing 11.2021: procedure and docu FL

### ISSUE THREAD

**Issue by floebe** [tira-io/tira#136]
## key resources
* https://www.tira.io/
* related
  * Integration Test column at [Agenda](https://github.com/tira-io/tira/projects/7) board
  * #132
  * #121 
* `authorship-verification-2021`   &nbsp;   https://www.tira.io/task/pan21-authorship-verification/user/test-user-01

## tasks & notes
* [o] ~~use testing material/MW's hints to do a test with the PAN baseline~~ -> superseded by other actions in parallel 
  * [x] clarify what exactly to do and how -> see TE hints in [testing material](#test-mat)

### <a name="test-mat">testing material</a>
* former PAN baselines   &nbsp;   https://github.com/pan-webis-de/pan-code
![grafik](https://user-images.githubusercontent.com/55153781/141497016-ac9a2ccd-431b-4a41-b65b-ff1d3e45b718.png)
* TE hints on that
  * use sub folder `clef21` for the tests
  * open task, `participate`, get VM running, connect to VM via button with access dates (below state)
  * ssh into vm, scp content from [clef21/style-change-detection](https://github.com/pan-webis-de/pan-code/tree/master/clef21/style-change-detection)

### misc
* test machine on Nov 11
  * http://betaweb001.medien.uni-weimar.de:31580/login

## log (summary)
* [x] Nov 12: getting access and initial test setup -> access to tira.io and vm works
* [x] Nov 13: initial testing, running into issues with `power on` and `shut down` -> led to
    * #140 
    * #141
* [x] Nov 14-18: repeated testing, with updates of the two issues and tentative new ones
* [x] Nov 18: major update of this issue (after major re-deployment(s) of TIRA)

**Comment by floebe** [comment]
first updates of this test documentation upon starting a bit of testing on Nov 13

**Comment by floebe** [comment]
updates of test documentation until before interruption at ca. 21:10

**Comment by floebe** [comment]
## history log Nov 12-13, all details
### Nov 12
* [x] try regular TIRA login FL -> works
* [x] info.discord >#tira: FL technically ready
* info MFröbe:
  * new test user created for TE, FL   &nbsp;    `test-user-01`
  * use any shared task to access the VM page -> FL tries `authorship-verification-2021`
* 13:15-30 VM page doesn't load / doesn't seem to be reachable -> **FL on hold**
* [x] clarify how to submit solutions in testing
  * [x] ~~@mam10eks docker containers available for that purpose? How to get & use?~~ -> no more reply?
  * MattiWe :  "redo some of the old pan baselines" (cf. testing material)

### Nov 13
* [x] check access to TIRA and vm -> works
  * new vm is `test-user-03`
  * host: `betaweb083.medien.uni-weimar.de`
* [x] ca. 18:45 check starting and stopping the VM at `authorship-verification-2021`
  * 1st attempt of `power on` -> for some seconds state is `powering on`, then again `stopped`
  * 2nd attempt of `power on` -> takes some seconds, then state `running` (SSH closed, RDP open)
  * 1st attempt of `shut down` -> for some few seconds state is `powering off`, then again `running`
  * 2nd attempt of `shut down` -> same as in 1st attempt
  * 3rd attempt of `shut down` -> same as in first two attempt, state remains `running`
  * having read [1] with the question "When you reload it's started and you get the correct state?", tried reloading the page -> state remains `running`
    [1] https://discord.com/channels/370896441863962626/640623874156920856/909020816824483870
  * [x] report this behavior on Discord
  * [x] file as issues
    * #140 
    * #141
* [x] try stopping the started vm `test-user-03` again, in any way -> led to new observations, to be repeated and reported
  * [x] try `stop vm` -> no effect at all, no state change at any time
  * [x] try `abort run` -> no visible effect at all, no state change at any time
  * [x] noticed that `stop vm` is highlighted when in `powering off` state, thus try `stop vm` in those seconds -> seemingly didn't work at first glance, clicked several times between `shut down` and `stop vm`; saw `powering off` for a short moment in between unexpectedly, then ` running` again, then at some point ` stopped`
  * [x] try to find a reliable work-around for reaching `stopped`  state -> unexpected observations
    * [x] restart vm -> hit `power on` once and try page reload -> state is now `running`
    * should update #140 with this observation
    * [x] try `shut down` on this properly started machine with page reload -> state is `stopped`
    * should update #141 with this observation
    * [x] repeat powering on and shutting down on `Author clustering` -> state after `power on` remains `stopped`, just as in #140 
    * [x] update #140 and #141 with new observations

interruption at 21:10
* [x] reproduce almost correct behavior on `authorship verification 2021` -> issues #140 and #141 so far remain; updates of them

**Comment by floebe** [comment]
## history log Nov 14: FL-1
* [x] check whether yesterday's problems [#140](#140) and [#141](#141) persist
  * 1st repeat on `authorship verification 2021` "successful" insofar that powering on required two times `power on`; but `shut down` led to `stopped` already on the third click on it
  * on a 2nd round at `authorship verification 2021`, `power on` works on 1st click + page reload, `shut down` on 1st click + page reload
  * question: just a matter of waiting?
* [x] try to repeat the behavior 
  * [x] on `author clustering`
    * directly got to: 1st click `power on` + page reload goes to `running`, then 1st click on `shut down` + page reload goes to `stopped`
  * [x] on `triple scoring`
    * 1st click on `power on` -> some seconds of `powering on`, then `stopped` + a short moment later `RDP port 55501` changes to `open`
    * page reload -> state is `running`, RDP port closed, `SSH port 44401` is `open`
    * 1st click on `shut down` -> some seconds of `powering off` (notably several more than on Nov 13), then state `running` + RDP and SSH ports `closed`, quickly thereafter state goes to `stopped' by itself
    * [x] on `Style Change Detection 2021`
      * 1st click on `power on` -> some seconds of `powering on`, then `stopped` and RDP and SSH ports remain `closed`
      * 4 page reloads (incl. some waiting time during 3. and 4.) don't change that picture
      * 2nd click on `power on` -> some seconds of `powering on`, then `stopped`, both ports `closed`
      * page reload -> state `running` and RDP port `open`
      * 1st click on `shut down` -> some seconds `powering off`, then no effect (`state `running` and RDP port `open`)
      * page reload -> no effect
      * 2nd click on `shut down` + 2 page reloads -> no effect
      * 3rd click on `shut down` + 4 page reloads -> no effect, but side observation: an overlay appears at the top navigation bar, of which I could only read "Loading VM ..." (perhaps "fails" as next word), then it disappears again
* [x] try stopping vm on `Style Change Detection 2021` in any way
  * [x] via `stop vm`
    * mouse over `stop vm` shows a hand as on a clickable link, but clicking has no effect
    * click on `shut down` and during state `powering off` click on `stop vm` (then highlighted as red button) -> state remains in `powering off`, then back to state `running` and RDP port `open`; a moment later RDP port changes to `closed`, state still `running`; another moment later state goes to `stopped`

**Comment by floebe** [comment]
### history log: Nov 14-Nov 18

* [x] move new observations into a logging comment in order to refer to them -> https://github.com/tira-io/tira/issues/136#issuecomment-968283992
* [x] update #140 with new observations: two types of behavior now
* [x] 14:30 re-check whether issues persist, given Bastian's [message at 14:00](https://discord.com/channels/370896441863962626/640623874156920856/909427689998401576) -> seems to remain
  * on `authorship verification 2021`: back to case 1 in #140 (I guess), at least `power on` does not work just once + page reload(s)
  * on `triple scoring`: behavior equal to `authorship verification`
* [x] update #141
* [x] inform via Discord
* [x] file question whether `stop vm` and `abort run` should always be clickable or only during specific states, e.g., `powering off` -> RE Nov 18 (tira-dev-meeting): yes, two kinds to abort only during `powering on` and `powering off`
* [x] ~~reproduce issues of `stop vm` and `abort run`, in order to report them as new GitHub-issues~~
* [x] ~~redo former "double start" on another task and check whether shutting down remains a problem~~

**Comment by MattiWe** [comment]
@floebe What needs to happen for this issue to be closed?

**Comment by floebe** [comment]
> @floebe What needs to happen for this issue to be closed?

Apologies for not noticing the comment yet (and overlooking the email message (in a bunch of further issue tracker messages)).

Nothing else is needed for this issue, the task of which got basically aborted at some point as the group fixed several issues in parallel - thus I close this one.
